aten/src/ATen/native/TensorShape.cpp

```diff
@@ -834,6 +836,25 @@ Tensor make_qtensor(const Tensor& self, IntArrayRef size, IntArrayRef stride, Qu
 }
 
 Tensor as_strided_tensorimpl(const Tensor& self, IntArrayRef size, IntArrayRef stride, optional<int64_t> storage_offset_) {
+
+
+  TensorImpl* tensor_impl = self.unsafeGetTensorImpl();
```

Comment:
can you explain how you decided to implement special behavior in as_strided_tensorimpl?  I'm not that familiar; at what point in the py script does this end up getting called?

Reply:
Yeah, the changes in TensorShape.cpp is because we are calling `print(loss)` outside of the lazy context.
Which flattens the tensor for the print out and goes through `as_strided_tensorimpl`.

Ideally, this part could be done automatically in the dispatcher (or alternatively, through an inter-operable LTCTensorImpl & TensorImpl), but that requires more code and I don't think it's necessary for our hack to go through the full thing.